nncf/torch/nncf_network.py

```diff
@@ -323,6 +323,42 @@ def __hash__(self):
         return hash(str(self))
 
 
+class KnowledgeDistillationLossHandler(nn.Module):
+    """
+    Encapsulates knowledge distillation logic. Controls loss calculation, its proper storing in case of parallel
+        execution on multiple devices and provides API for external knowledge distillation algorithm.
+    """
+    KD_LOSS_STORAGE_NAME = 'kd_loss'
+    KD_STORAGE_DEVICE = 'kd_storage_device'
+    def __init__(self, context, kd_original_model, calculate_kd_loss_fn, storage_device):
+        super().__init__()
+        self._compressed_context = context
+        self.is_enabled = False
+        self._kd_original_model = kd_original_model
+        self._calculate_kd_loss_fn = calculate_kd_loss_fn
+        self._compressed_context.register_global_buffer(self.KD_LOSS_STORAGE_NAME, [])
+        self._compressed_context.register_global_buffer(self.KD_STORAGE_DEVICE, storage_device)
+
+    def zero_kdloss(self):
+        if self.is_enabled is not None:
+            self._compressed_context.global_buffer_store[self.KD_LOSS_STORAGE_NAME] = []
+
+    def get_kdloss(self):
+        if self.is_enabled is not None:
+            return self._compressed_context.global_buffer_store[self.KD_LOSS_STORAGE_NAME]
+        return 0
```

Comment:
Type hints. Make sure that the PyTorch int `0` is what you actually want to be returned - shouldn't this be `torch.Tensor` instead?